### PR TITLE
Building scripting + grunt integration

### PR DESCRIPTION
**Pull request by wouldgo**
Little summary:
- As reported in #67 I refactored the building/cleaning scripts (I didn't receive any error now).
- I introduced grunt, with two task for this first version: jshint and tape runners.

Waiting for feedbacks for pull request eligibility.

Greetings, 
Dario.

### CODE REVIEW

**Comment by piranna** [comment]
Good job :-) Just to note: since we are now using grunt, we should use grunt-gyp and grunt-contrib-clean plugins.

**Comment by wouldgo** [comment]
No problem using these tasks.
As soon as I complete this week iteration at work I'll able to introduce and configure :)

**Comment by piranna** [comment]
Cool then :-)

**Comment by modeswitch** [comment]
There's some overlap here with #69.

**Comment by wouldgo** [comment]
From a brief review (I'm on train returning home) the differences between mine and his are mainly the fact I use make as compilation tool (since I found that sometimes ninja fails due a python problem in import) and I use some options for gclient unused in the source build.js.

Waiting some feedbacks considerations.

**Comment by DamonOehlman** [comment]
Hey All,

I guess the approach I was taking in #69 was to look at trying to get us some way down the path towards being able to have the module build on Windows (see #42).  From the perspective of running gclient and bootstrapping the build it definitely seems to work on Windows - I gave it a run on  a Windows surface tablet last night.  There is still a little way to go but I can successfully compile all the libwebrtc dependencies so that's a good start.

From a sense of reviewing this PR, there seems to be a lot in it so it's difficult to look at the install build script in isolation.  So my suggestion would be to that if @modeswitch is happy with the general direction and content of this PR to merge it and I'll revisit #69 again down the track if it building on windows becomes a higher priority.  I'm not on Windows so I'm happy to wait :)

Cheers,
Damon.

**Comment by modeswitch** [comment]
There's a lot happening in this patch. It would be helpful to split it out into smaller pieces.

@DamonOehlman I've put windows support in Milestone 0.2, so I'm not rushed to get that in.

**Comment by DamonOehlman** [comment]
@modeswitch Cool - I think that's wise given what I had to go through last night.  I followed the [Building Chromium with Visual Studio Express](http://www.chromium.org/developers/how-tos/build-instructions-windows#TOC-Setting-up-the-environment-for-building-with-Visual-C-2010-Express-or-Windows-7.1-SDK) instructions and I think I probably downloaded half the Internet getting all the dependencies...

**Comment by wouldgo** [comment]
@modeswitch except the build.js+clean.js and grunt there isn't so much in this patch (all the other modifications are due jshint complains).

Anyway I'm open to review my work.

**Comment by modeswitch** [comment]
@wouldgo Could you split the jshint work into a separate pull?

**Comment by wouldgo** [comment]
done.

File: binding.gyp [hunk 1]
```diff
@@ -19,7 +19,7 @@
       'target_name': 'webrtc',
       'dependencies': [],
       'variables': {
-        'libwebrtc_out%': '<(libwebrtc)/out/Release/obj',
+        'libwebrtc_out%': '<(libwebrtc)/out/Release/obj.target',
```
Comment (modeswitch) [inline binding.gyp:22]: What's the purpose of this?

File: package.json [hunk 2]
```diff
@@ -19,11 +23,16 @@
...
+    "tape": "~2.4.2",
     "ws": "~0.4.31",
-    "tape": "~2.4.2"
+    "grunt": "~0.4.2",
+    "grunt-contrib-jshint": "~0.6.*",
+    "grunt-contrib-uglify": "~0.2.4",
+    "grunt-tape": "~0.0.2"
   },
   "scripts": {
     "install": "node bin/build.js",
```
Comment (modeswitch) [inline package.json:38]: Since we're now using Grunt, can you convert the build and clean scripts into grunt tasks?

**Comment by DamonOehlman** [comment]
I'd probably advise against using grunt for the actual build and clean scripts themselves.  I think grunt is an excellent tool for helping workflow build tasks that should be undertaken by a maintainer or contributors, but it's not something that people installing the module should have to install as a "non dev" dependency.  Especially considering that for the build to run you'd need to include the `grunt-cli` package also in dependencies.

I realize the original intention of #56 was to probably use grunt in this way, but I guess I'd suggest taking a moment to decide whether that's really what you want.  If it is - then all good :)

**Comment by wouldgo** [comment]
So:
- I've modified the bindings.gyp since make creates an out.target folder instead of out;
- I agree with @DamonOehlman about the grunt considerations. I see that modules as an helper for development tasks (like testing, linting and rebuilding c++ bindings).

**Pull request merged**